### PR TITLE
Created Product - parser, controller and service

### PR DESCRIPTION
**Pull request by himanshu-nodex**

### CODE REVIEW

File: expressEngine/lib/parser/base.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Parser {
+  error = (err) => {
+    const errMsg =
+      (err.errors && err.errors && err.errors[Object.keys(err.errors)[0]].message) ||
```
Comment (manish-nx) [inline expressEngine/lib/parser/base.js]: duplicate check for err.errors

File: expressEngine/cms/product/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import ErrorMsgs from './error';
+import InfoMsgs from './info';
+import SuccessMsgs from './success';
```
Comment (manish-nx) [inline expressEngine/cms/product/index.js]: replace
ErrorMsgs to error 
InfoMsgs to info
SuccessMsgs to success

File: expressEngine/cms/product/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+export default { ...ErrorMsgs, ...InfoMsgs, ...SuccessMsgs };
```
Comment (manish-nx) [inline expressEngine/cms/product/index.js]: unnecessary use of spread operator here.
Replace
export default { ...ErrorMsgs, ...InfoMsgs, ...SuccessMsgs }; to export default { errror, info, success };

File: expressEngine/product/controller/product.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// import base controller
+import { BaseController } from '../../lib/controller/index';
+
+// import service
+import { ProductService } from '../service/index';
+
+class ProductController extends BaseController  {
+  create = async ({body}) => {
+    console.log("#####################", body);
```
Comment (manish-nx) [inline expressEngine/product/controller/product.js]: remove console.log

File: expressEngine/product/parser/product.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// import the controller
```
Comment (manish-nx) [inline expressEngine/product/parser/product.js]: remove this comment

File: expressEngine/product/parser/product.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import { BaseParser } from '../../lib/parser/index';
+
+class ProductParser extends BaseParser {
+
+  run(controller, functionName){
+
+    return async (req, res)  => {
+      const { body, params, query } = req;
+      try {
+        const result = await controller[functionName]({ params, query, body });
```
Comment (manish-nx) [inline expressEngine/product/parser/product.js]: add check for controller && controller.length

File: expressEngine/product/service/product.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// import service libraries
+import { BaseService, DbService } from '../../lib/service/index';
```
Comment (manish-nx) [inline expressEngine/product/service/product.js]: replace DbService to DBService

File: expressEngine/product/service/product.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+// import collections
+import { ProductModel, ProductDetail } from '../../model/index';
+
+// import messages
+import cms from '../../cms/product/index';
+
+class Service extends BaseService {
+  registerProduct = async (data) => {
+    try {
+      // const data = { ...req.body };
```
Comment (manish-nx) [inline expressEngine/product/service/product.js]: remove this comment

File: expressEngine/product/service/product.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      this.validateRequired(data, requiredFields);
+
+      const isExist = await DbService.count(ProductModel, { name: data.name, supplierId: data.supplierId });
+      if (isExist) {
+        return this.error(cms.alreadyRegistered);
+      }
+
+
+      const Product = await DbService.create(ProductModel, {
+        name: data.name,
```
Comment (manish-nx) [inline expressEngine/product/service/product.js]: add checks like data && data.name

File: expressEngine/user/controller/user.js [hunk 10]
```diff
@@ -4,16 +4,16 @@ import { BaseController } from '../../lib/controller/index';
 // import service
 import { UserService } from '../service/index';
 
-class Controller extends BaseController {
-  create = async (req, res) => {
+class UserController extends BaseController {
```
Comment (manish-nx) [inline expressEngine/user/controller/user.js]: replace UserController to Controller.

File: expressEngine/user/parser/user.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+// import base controller
+import { BaseParser } from '../../lib/parser/index';
+
+class Parser extends BaseParser {
+
+  create = async (req, res)  => {
+    try {
+      const { email, password } = req.body;
+      const result = await UserController.create({ email, password });
```
Comment (manish-nx) [inline expressEngine/user/parser/user.js]: add error check here.

File: expressEngine/user/service/user.js [hunk 12]
```diff
@@ -8,22 +8,29 @@ import { User, UserDetail } from '../../model/index';
...
-    const requiredFields = ["email", "password"];
-    this.validateRequired(data, requiredFields);
-
-    const isExist = await DbService.count(User, { email: data.email });
-    if (isExist) {
-      return this.response(res, this.error(cms.alreadyRegistered), 409);
-    }
+  registerUser = async (data) => {
+    try {
+      // const data = { ...req.body };
```
Comment (manish-nx) [inline expressEngine/user/service/user.js]: remove this comment

File: expressEngine/product/service/product.js (hunk 9 above)
Comment (amanjotSuccessive) [reply expressEngine/product/service/product.js]: DbService is correct notation. It doesn't need to be changed

File: expressEngine/product/service/product.js (hunk 7 above)
Comment (amanjotSuccessive) [reply expressEngine/product/service/product.js]: DbService is correct notation. It doesn't need to be changed

**Pull request merged**